core/commonMain/src/extensions.kt

```diff
@@ -766,3 +806,107 @@ public fun <K, V> Map<K, V>.toPersistentHashMap(): PersistentMap<K, V>
         = this as? PersistentHashMap
         ?: (this as? PersistentHashMapBuilder<K, V>)?.build()
         ?: PersistentHashMap.emptyOf<K, V>().putAll(this)
+
+
+/**
+ * Returns a persistent map of all elements in this [Sequence] of [Pair]s,
+ * using [Pair.first] as [keys][ImmutableMap.keys] and [Pair.second] as [values][ImmutableMap.values].
+ *
+ * @since 0.4
+ */
+public fun <T, R> Sequence<Pair<T, R>>.toPersistentMap(): PersistentMap<T, R> =
```

Comment:
Could you please add tests for the new functions?

Reply:
Not really; Idk how the current functions are tested, so I can't know how you want the tests to be done. I could not find these tests myself.

Reply:
`Map` tests are mostly in this file: https://github.com/Kotlin/kotlinx.collections.immutable/blob/master/core/commonTest/src/contract/map/ImmutableMapTest.kt

You can add tests for new functions either there or in a new file within the directory.